File: Data Structures & Algorithms/anagram-groups/submission-0.cs
public class Solution {
    public List<List<string>> GroupAnagrams(string[] strs) {
        int[] counts = new int[26];
        var dict = new Dictionary<string, List<string>>();
        foreach(string s in strs){
            Array.Clear(counts, 0, counts.Length);
            foreach(char c in s){
                counts[c - 'a']++;
            }

            string key = string.Join(',', counts);

            if(!dict.ContainsKey(key)){
                dict[key] = new List<string>();
            }
            dict[key].Add(s);
            
        }
        var res = new List<List<string>>();
        foreach(var pair in dict){
            res.Add(pair.Value);
        }
        return res;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res = ""
        separator = "|"
        for s in strs:
            count = len(s)
            res += str(count) + separator + s
        return res;

    def decode(self, s: str) -> List[str]:
        res = []
        separator = "|"
        i = 0
        while i < len(s):
            j = i
            while s[j] != separator:
                j+=1
                l = int(s[i:j])
            res.append(s[j+1:j+1+l])
            i = j+1+l
        return res;